pytorch_lightning/accelerators/cpu.py

```diff
@@ -29,8 +29,10 @@ def setup(self, trainer: "pl.Trainer") -> None:
             MisconfigurationException:
                 If the selected device is not CPU.
         """
-        if "cpu" not in str(self.root_device):
-            raise MisconfigurationException(f"Device should be CPU, got {self.root_device} instead.")
+        if "cpu" not in str(self.training_type_plugin.root_device):
+            raise MisconfigurationException(
+                f"Device should be CPU, got {self.training_type_plugin.root_device} instead."
+            )
```

Comment:
do you think these checks should sit in the strategy after the refactor? the accelerator won't have a reference to training type soon.

Reply:
Yeah, it will happen in step3 "move accelerator/device" to strategy with other misconfig checks.
I think it make more sense to do in step3, as when we move all accelerator.setup() login into strategy. There are other misconfi checks like what strategy/precision tpu should use.